controllers/internal/utils/dns.go

```diff
@@ -20,30 +20,92 @@ package utils
 import (
 	"fmt"
 	"sort"
+	"strings"
 
 	"github.com/lixiangzhong/dnsutil"
+	"github.com/miekg/dns"
 )
 
-// Dig retrieves list of tuple <IP address, A record > from edge DNS server for specific FQDN
-func Dig(edgeDNSServer, fqdn string) ([]string, error) {
+type DNSServer struct {
+	Host string
+	Port int
+}
+
+type DNSList []DNSServer
+
+func (s DNSServer) String() string {
+	return fmt.Sprintf("%s:%v", s.Host, s.Port)
+}
+
+func (l DNSList) String() string {
+	var aux []string
+	for _, el := range l {
+		aux = append(aux, el.String())
+	}
+	return strings.Join(aux, ",")
+}
+
+// Dig returns list of tuple <IP address, A record > for a given FQDN by using the dns servers from edgeDNSServers
+// dns servers are tried taking always two of them (setting one as a primary and the other as a backup one)
```

Comment:
not sure here, the result is the faster one, see https://github.com/lixiangzhong/dnsutil/blob/e61fde722b0a86b89243c686efc8c32fac5223d0/dig.go#L127-L129